src/core/net/dnssd_server.cpp

```diff
@@ -1278,9 +1287,11 @@ void Server::ResetUpstreamQueryTransaction(UpstreamQueryTransaction &aTxn, Error
     if (aError == kErrorNone)
     {
         LogInfo("Upstream query transaction %d completed.", index);
+        mCounters.mUpstreamDnsCounters.mResponses++;
```

Comment:
Let's be consistent on 'log before or after adding the counters'. (In the else branch, counter is added before logging while here counter is added after logging). I prefer adding counters first and then do the logging.